### PR TITLE
Add no-value key handling only for form body

### LINKED ISSUE #13981: JSON in POST request treated as body attribute in HttpPostStandardRequestDecoder
### Expected behavior

JSON in POST request shouldn't be counted as body attribute in `HttpPostStandardRequestDecoder`, as in releases `<4.1.108-Final` (checked on `4.1.107.Final` and `4.1.106.Final`)

### Actual behavior

JSON in POST request is treated as body attribute in `HttpPostStandardRequestDecoder`, and an attribute is created with key equal to JSON and empty value

### Steps to reproduce

Send a POST request to netty `>=4.1.108-Final`, decode the request using `HttpPostRequestDecoder` and check `.getBodyHttpDatas() ` The resulting list will contain Attribute HttpData where the key is a JSON sent by a client, and value is empty

### Minimal yet complete reproducer code (or URL to code)

```
public class NettyReproduction {
    public static void main(String[] args) {
        FullHttpRequest request = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.POST, "/netty/reproduction",
            Unpooled.wrappedBuffer("""
                {
                    "netty": "reproduction"
                }
                """.getBytes(StandardCharsets.UTF_8)));
        var decoder = new HttpPostRequestDecoder(request);
        var result = decoder.getBodyHttpDatas();
        System.out.println(result); 
    }
}
```
results in:
`[Mixed: {    "netty": "reproduction"}=]` on `4.1.109.Final` and `4.1.108.Final` 

but results in:
`[]` on `4.1.107.Final` and `4.1.106.Final`

### Netty version
as above

### JVM version (e.g. `java -version`)
Java 21

### OS version (e.g. `uname -a`)
Windows 10, 
and @Rufusmen reproduced on Linux (Ubuntu 22.04.4 LTS x86_64)

We suspect that the reason for such change is this PR: https://github.com/netty/netty/pull/13908 , thus we are wondering if such behaviour is desired, if the decoder decodes something different than `x-www-form-urlencoded`

### PR DESCRIPTION
**Pull request by gniadeck**
Motivation:

This is a fix for issue #13981 that reports a changed behaviour of HttpPostStandardRequestDecoder after this PR - https://github.com/netty/netty/pull/13908 

Because HttpPostStandardRequestDecoder changed the contract, some code implementations relying on certain parsing are failing

Modification:

This PR makes sure, that the edge case handling for form body happenes only when the content is in fact form body

Result:

Fixes #13981

### CODE REVIEW

File: codec-http/src/test/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoderTest.java [hunk 1]
```diff
@@ -17,16 +17,11 @@
...
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.Unpooled;
-import io.netty.handler.codec.http.DefaultHttpContent;
-import io.netty.handler.codec.http.DefaultHttpRequest;
-import io.netty.handler.codec.http.DefaultLastHttpContent;
-import io.netty.handler.codec.http.HttpMethod;
-import io.netty.handler.codec.http.HttpRequest;
-import io.netty.handler.codec.http.HttpVersion;
-import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.handler.codec.http.*;
```
[the remark this answers is not in the document]
Comment (gniadeck) [reply codec-http/src/test/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoderTest.java]: fixed :)

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java [hunk 2]
```diff
@@ -783,6 +784,17 @@ public void removeHttpDataFromClean(InterfaceHttpData data) {
         factory.removeHttpDataFromClean(request, data);
     }
 
+    /**
+     * Check if request has headers indicating that it contains form body
+     */
+    private boolean hasFormBody() {
+        String contentHeader = request.headers().get("Content-Type");
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java:791]: Use static HttpHeaderNames value

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java [hunk 3]
```diff
@@ -783,6 +784,17 @@ public void removeHttpDataFromClean(InterfaceHttpData data) {
...
+        if (contentHeader == null) {
+            return false;
+        }
+        return contentHeader.equals("application/x-www-form-urlencoded") || contentHeader.equals("multipart/form-data");
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java:795]: Use static HttpHeaderValues values

**Review by normanmaurer — CHANGES_REQUESTED** [review]

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java (hunk 2 above)
Comment (gniadeck) [reply codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java]: fixed, also renamed the `contentHeader` variable to `contentHeaderValue` :)

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java (hunk 3 above)
Comment (gniadeck) [reply codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java]: fixed :)

**Review by normanmaurer — APPROVED** [review]

**Comment by normanmaurer** [comment]
@gniadeck thanks!

**Comment by yawkat** [comment]
@gniadeck can I ask what the use case is, here? I'm working on rewriting the post body parsing ( https://github.com/netty-contrib/codec-multipart/pull/25 ), so I'd like to understand whether I need to support non-standard parsing like this

**Comment by chrisvest** [comment]
@gniadeck Can you take a look at Jonas' comments?

**Comment by yawkat** [comment]
We can probably leave it like this but I won't port this PR to
contrib-multipart.
